catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/inputtransformer/InputTransformerProducer.java

```diff
@@ -131,64 +144,43 @@ private Metacard generateMetacard(
-      // can create the metacard, then do not need to try any remaining InputTransformers.
-      for (InputTransformer transformer : listOfCandidates) {
-
-        try (InputStream inputStreamMessageCopy =
-            fileBackedOutputStream.asByteSource().openStream()) {
-          if (StringUtils.isEmpty(metacardId)) {
-            generatedMetacard = transformer.transform(inputStreamMessageCopy);
-          } else {
-            generatedMetacard = transformer.transform(inputStreamMessageCopy, metacardId);
-          }
-        } catch (IOException | CatalogTransformerException e) {
-          LOGGER.debug("Transformer [{}] could not create metacard.", transformer, e);
-        }
-        if (generatedMetacard != null) {
-          break;
+    // Multiple InputTransformers may be found that match the mime type.
+    // Need to try each InputTransformer until we find one that can successfully transform
+    // the input stream's data into a metacard. Once an InputTransformer is found that
+    // can create the metacard, then do not need to try any remaining InputTransformers.
+    for (InputTransformer transformer : listOfCandidates) {
+
+      try (InputStream inputStreamMessageCopy = tfbos.asByteSource().openStream()) {
+        if (StringUtils.isEmpty(metacardId)) {
+          generatedMetacard = transformer.transform(inputStreamMessageCopy);
+        } else {
+          generatedMetacard = transformer.transform(inputStreamMessageCopy, metacardId);
         }
+      } catch (CatalogTransformerException e) {
+        LOGGER.debug("Transformer [{}] could not create metacard.", transformer, e);
+      } catch (IOException e) {
+        LOGGER.debug("Could not open input stream", e);
       }
-
-      if (generatedMetacard == null) {
-        throw new MetacardCreationException(
-            "Could not create metacard with mimeType "
-                + mimeType
-                + ". No valid transformers found.");
+      if (generatedMetacard != null) {
+        break;
       }
+    }
 
-      LOGGER.trace("EXITING: generateMetacard");
-    } catch (IOException e) {
-      throw new MetacardCreationException("Could not create metacard.", e);
+    if (generatedMetacard == null) {
+      return Optional.empty();
     }
 
-    return generatedMetacard;
+    LOGGER.trace("EXITING: generateMetacard");
+    return Optional.of(generatedMetacard);
   }
 
-  private String getMimeTypeFromHeader(Message in, InputStream is) {
-    String fileExtension = getHeaderAsStringAndRemove(in, FILE_EXTENSION_HEADER);
-    if (fileExtension == null) {
-      return null;
-    }
-
+  private Optional<String> getMimeTypeFor(InputStream is, String fileExtension) {
     try {
-      return ((CatalogEndpoint) getEndpoint()).getMimeTypeMapper().guessMimeType(is, fileExtension);
+      return Optional.of(
+          ((CatalogEndpoint) getEndpoint()).getMimeTypeMapper().guessMimeType(is, fileExtension));
```

Comment:
❓ Can `guessMimeType()` return `null`? If so then use `Optional.ofNullable()`

Reply:
Yes it can, good catch.